pyttsx3/drivers/avsynth.py

```diff
@@ -0,0 +1,143 @@
+import objc
+from AppKit import NSSpeechSynthesizer
+from Foundation import NSObject, NSTimer, NSRunLoop, NSDefaultRunLoopMode, NSURL, NSDate
+from PyObjCTools import AppHelper
+from ..voice import Voice
+
+
+# noinspection PyPep8Naming
+def buildDriver(proxy):
+    driver = AVSpeechDriver.alloc().init()
+    driver.setProxy(proxy)
+    driver.initialize_busy_state()
+    return driver
+
+
+class AVSpeechDriver(NSObject):
+    def init(self):
+        self = objc.super(AVSpeechDriver, self).init()
+        if self is None:
+            return None
+        self._proxy = None
+        self._tts = NSSpeechSynthesizer.alloc().initWithVoice_(None)
+        self._tts.setDelegate_(self)
+        self._current_voice = None
+        self._queue = []
+        self._should_stop_loop = False  # Flag to control loop
+        return self
+
+    @objc.python_method
+    def setProxy(self, proxy):
+        self._proxy = proxy
+
+    @objc.python_method
+    def initialize_busy_state(self):
+        if self._proxy:
+            self._proxy.setBusy(False)
+
+    def destroy(self):
+        self._tts.setDelegate_(None)
+        self._tts = None
+
+    def onPumpFirst_(self, timer):
+        self._proxy.setBusy(False)
+
+    def startLoop(self):
+        self._should_stop_loop = False
+        NSTimer.scheduledTimerWithTimeInterval_target_selector_userInfo_repeats_(
+            0.0, self, "processQueue:", None, True
+        )
+        AppHelper.runConsoleEventLoop()
+
+    @objc.signature(b"v@:@")
```

Comment:
pytest warning:
```
tests/test_pyttsx3.py::test_speaking_text
  /Users/cclauss/Python/pyttsx3/pyttsx3/drivers/avsynth.py:52:
   DeprecationWarning: Usage objc.typedSelector instead of objc.signature
    @objc.signature(b"v@:@")
```
https://pyobjc.readthedocs.io/en/latest/api/module-objc.html#objc.typedSelector
```suggestion
    @objc.typedSelector(b"v@:@")
```